plugin-test/src/test/groovy/org/gradle/github/dependency/extractor/BaseExtractorTest.groovy

```diff
@@ -7,43 +7,37 @@ import groovy.json.JsonSlurper
 import groovy.transform.CompileDynamic
 import groovy.transform.CompileStatic
 import groovy.transform.Memoized
-import org.gradle.test.fixtures.SimpleGradleExecuter
 import org.gradle.github.dependency.fixture.TestConfig
 import org.gradle.internal.hash.Hashing
-import org.gradle.test.fixtures.build.BuildTestFile
-import org.gradle.test.fixtures.build.BuildTestFixture
+import org.gradle.test.fixtures.SimpleGradleExecuter
 import org.gradle.test.fixtures.file.TestFile
-import org.gradle.test.fixtures.file.TestNameTestDirectoryProvider
 import org.gradle.test.fixtures.maven.MavenFileRepository
 import org.gradle.testkit.runner.BuildResult
 import org.gradle.util.GradleVersion
-import org.intellij.lang.annotations.Language
-import org.junit.Rule
 import spock.lang.Specification
+import spock.lang.TempDir
 
 import java.util.stream.Collectors
 
 abstract class BaseExtractorTest extends Specification {
 
-    @Rule
-    public final TestNameTestDirectoryProvider temporaryFolder = new TestNameTestDirectoryProvider(getClass())
+    @TempDir
+    File testDir
```

Comment:
If that becomes a problem, we can deal with it. I'm not a fan of all of the code borrowed from the Gradle Build Tool project, because we end up with a lot of unused code in the codebase.